xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -677,31 +658,31 @@ bool CPVRChannelGroup::RemoveDeletedChannels(const CPVRChannelGroup &channels)
   CSingleLock lock(m_critSection);
 
   /* check for deleted channels */
-  for (int iChannelPtr = m_members.size() - 1; iChannelPtr >= 0; iChannelPtr--)
+  for (std::vector<PVRChannelGroupMember>::iterator it = m_members.begin(); it != m_members.end();)
   {
-    CPVRChannelPtr channel = m_members.at(iChannelPtr).channel;
-    if (!channel)
-      continue;
-
-    if (channels.GetByClient(channel->UniqueID(), channel->ClientID()) == NULL)
+    if (channels.GetByClient((*it).channel->UniqueID(), (*it).channel->ClientID()) == NULL)
     {
       /* channel was not found */
       CLog::Log(LOGINFO,"PVRChannelGroup - %s - deleted %s channel '%s' from group '%s'",
-          __FUNCTION__, m_bRadio ? "radio" : "TV", channel->ChannelName().c_str(), GroupName().c_str());
+          __FUNCTION__, m_bRadio ? "radio" : "TV", (*it).channel->ChannelName().c_str(), GroupName().c_str());
 
       /* remove this channel from all non-system groups if this is the internal group */
       if (IsInternalGroup())
       {
-        g_PVRChannelGroups->Get(m_bRadio)->RemoveFromAllGroups(channel);
+        g_PVRChannelGroups->Get(m_bRadio)->RemoveFromAllGroups((*it).channel);
 
         /* since it was not found in the internal group, it was deleted from the backend */
-        channel->Delete();
+        (*it).channel->Delete();
       }
 
-      m_members.erase(m_members.begin() + iChannelPtr);
+      m_members.erase(it++);
       m_bChanged = true;
```

Comment:
and thanks for reminding me to check all of these `erase()` calls another time, because I've just found another one that I had missed :)
replacing the `erase(it++)` calls by `it = erase(it)` in the files that I'm editing now so we don't need 10 posts on github explaining it ;-)